crypto/s2n_ecc_evp.c

```diff
@@ -139,7 +148,8 @@ static int s2n_ecc_evp_generate_own_key(const struct s2n_ecc_named_curve *named_
         return s2n_ecc_evp_generate_key_x25519(named_curve, evp_pkey);
     }
 #endif
-    if (named_curve->libcrypto_nid == NID_X9_62_prime256v1 || named_curve->libcrypto_nid == NID_secp384r1) {
+    if (named_curve->libcrypto_nid == NID_X9_62_prime256v1 || named_curve->libcrypto_nid == NID_secp384r1
+        || named_curve->libcrypto_nid == NID_secp521r1) {
```

Comment:
NitPick: I really feel like `s2n_ecc_named_curve` should have a new field called `generate_key` of (rough) type `S2N_RESULT(const struct s2n_ecc_named_curve *named_curve, EVP_PKEY **evp_pkey)`

Reply:
update!